src/monodroid/jni/android-system.cc

```diff
@@ -203,10 +203,20 @@ AndroidSystem::_monodroid__system_property_get (const char *name, char *sp_value
 	if (env_value_len >= sp_value_len)
 		log_warn (LOG_DEFAULT, "System property buffer size too small by %u bytes", env_value_len == sp_value_len ? 1 : env_value_len - sp_value_len);
 
-	strncpy (sp_value, env_value, sp_value_len);
+	//
+	// sp_value_len includes the terminating nul, avoid a mingw g++ warning about string truncation
+	// by making the amount of data copied one less than the indicated length. The warning reported
+	// is:
+	//
+	//  In function ‘int xamarin::android::internal::AndroidSystem::_monodroid__system_property_get(const char*, char*, size_t)’,
+	//    inlined from ‘int xamarin::android::internal::AndroidSystem::monodroid_get_system_property(const char*, char**)’ at ../../../jni/android-system.cc:243:44:
+	//    ../../../jni/android-system.cc(206,10): warning G20816D19: ‘char* strncpy(char*, const char*, size_t)’ specified bound 93 equals destination size [-Wstringop-truncation] [/home/grendel/vc/xamarin/xamarin-android-worktrees/code-quality-improvements/src/monodroid/monodroid.csproj]
+	//    strncpy (sp_value, env_value, sp_value_len);
+	//
+	strncpy (sp_value, env_value, sp_value_len - 1);
 	sp_value[sp_value_len] = '\0';
 
-	return strlen (sp_value);
+	return static_cast<int>(strlen (sp_value));
```

Comment:
I think we can change the return type here to `size_t` and change above check `if (!name || !sp_value) return -1;` to return empty string (`sp_value[0] = '\0'; return 0;`) instead.

The `-1` check might be done in the calling functions, where it would not be needed (the pointers are not null) or simplified (only one comes from outside as parameter).

Reply:
While I like the idea, the function is reimplementation of the Android's `__system_property_get` which we wrap in our Android-specific code, and it returns an `int` with `-1` signalling an error - I think it's better to keep compatibility here.

Reply:
`__system_property_get` has different signature anyway. so we might just rename `_monodroid__system_property_get` to something like `_monodroid_get_system_property` and make the returned length type be in line with other `monodroid_get_system_property*` functions?

Reply:
We need a way to signal an error, though (in case either, or both, `name` or `sp_value` are `nullptr`) - in this case we currently return `-1`. `0` is a valid length, so we can't use that as the error indicator.

Reply:
Yesterday I misread your patch and thought `monodroid_get_system_property` return type changed to `size_t`, which is actually not the case. Only the `monodroid_get_system_property_*` changed. I wonder whether they should not return `-1` as well in case the property is not available.

Reply:
Yep, they probably should, but that's beyond the scope of this PR :)

Reply:
We might at least keep them returning an `int` then.